RecoLocalTracker/SiPixelClusterizer/python/siPixelClustersPreSplitting_cff.py

```diff
@@ -93,3 +94,94 @@
                             siPixelDigisClustersPreSplitting,
                             # SwitchProducer wrapping the legacy pixel cluster producer or an alias for the pixel clusters information converted from SoA
                             siPixelClustersPreSplitting))
+
+######################################################################
+
+### Alpaka Pixel Clusters Reco
+
+#from CalibTracker.SiPixelESProducers.siPixelCablingSoAESProducer_cfi import siPixelCablingSoAESProducer
+#from CalibTracker.SiPixelESProducers.siPixelGainCalibrationForHLTSoAESProducer_cfi import siPixelGainCalibrationForHLTSoAESProducer
+
+def _addProcessCalibTrackerAlpakaES(process):
+    process.load("CalibTracker.SiPixelESProducers.siPixelCablingSoAESProducer_cfi")
+    process.load("CalibTracker.SiPixelESProducers.siPixelGainCalibrationForHLTSoAESProducer_cfi")
+
+modifyConfigurationCalibTrackerAlpakaES_ = alpaka.makeProcessModifier(_addProcessCalibTrackerAlpakaES)
+
+# reconstruct the pixel digis and clusters with alpaka on the device
+from RecoLocalTracker.SiPixelClusterizer.siPixelRawToClusterPhase1_cfi import siPixelRawToClusterPhase1 as _siPixelRawToClusterAlpaka
+siPixelClustersPreSplittingAlpaka = _siPixelRawToClusterAlpaka.clone()
+
+(alpaka & run3_common).toModify(siPixelClustersPreSplittingAlpaka,
+    # use the pixel channel calibrations scheme for Run 3
+    clusterThreshold_layer1 = 4000,
+    VCaltoElectronGain      = 1,  # all gains=1, pedestals=0
+    VCaltoElectronGain_L1   = 1,
+    VCaltoElectronOffset    = 0,
+    VCaltoElectronOffset_L1 = 0)
+
+from RecoLocalTracker.SiPixelClusterizer.siPixelPhase2DigiToCluster_cfi import siPixelPhase2DigiToCluster as _siPixelPhase2DigiToCluster
+
+(alpaka & phase2_tracker).toReplaceWith(siPixelClustersPreSplittingAlpaka, _siPixelPhase2DigiToCluster.clone(
+    Phase2ReadoutMode = PixelDigitizerAlgorithmCommon.Phase2ReadoutMode.value(), # flag to decide Readout Mode : linear TDR (-1), dual slope with slope parameters (+1,+2,+3,+4 ...) with threshold subtraction
+    Phase2DigiBaseline = int(PixelDigitizerAlgorithmCommon.ThresholdInElectrons_Barrel.value()), # same for barrel and endcap
+    Phase2KinkADC = 8,
+    ElectronPerADCGain = PixelDigitizerAlgorithmCommon.ElectronPerAdc.value()
+))
+
+# reconstruct the pixel digis and clusters with alpaka on the cpu, for validation
+siPixelClustersPreSplittingAlpakaSerial = siPixelClustersPreSplittingAlpaka.clone(
+    #alpaka = dict( backend = '*' )
+    alpaka = None
+)
+siPixelClustersPreSplittingAlpakaSerial._TypedParameterizable__type = 'alpaka_serial_sync' + siPixelClustersPreSplittingAlpaka._TypedParameterizable__type.removesuffix('@alpaka')
```

Comment:
Yes, this is ugly, but it's the only way I found to make it work in the case where a GPU backend is requested explicitly.

Reply:
To add more details: the main use for such a module is to run the GPU-vs-CPU validation, where it makes sense to require `process.options.accelerators = 'gpu-nvidia'.

The problem is that using an `@alpaka` module with an explicit `alpaka.backend = 'serial_sync'` with `process.options.accelerators = 'gpu-nvidia'` will fail:
```
Exception Message:
Module siPixelClustersPreSplittingAlpakaSerial has the Alpaka backend set explicitly, but its accelerator is not available for the job because of the combination of the job configuration and accelerator availability on the machine. The following Alpaka backends are available for the job cuda_async.
```

Reply:
If this approach works, I would suggest to write a helper function that takes an `@alpaka` module, clones it, and changes it to use explicitly the `alpaka_serial_sync` backend.

Reply:
Hmh, my first thought is that I'd like the `alpaka.backend = 'serial_sync'` to do the job in this case, but I'd need a more time to think through the interplay with `process.options.accelerators`. Could you open an issue about this question?

Reply:
https://github.com/cms-sw/cmssw/issues/43780